lambdas/transcode/index.py

```diff
@@ -0,0 +1,95 @@
+"""
+Generate video previews for videos in S3.
+"""
+import subprocess
+import tempfile
+
+from t4_lambda_shared.decorator import api, validate
+from t4_lambda_shared.utils import get_default_origins, make_json_response
+
+SCHEMA = {
+    'type': 'object',
+    'properties': {
+        'url': {
+            'type': 'string'
+        },
+        'format': {
+            'enum': ['mp4', 'webm']
+        },
+        'width': {
+            'type': 'string'
+        },
+        'height': {
+            'type': 'string'
+        },
+        'duration': {
+            'type': 'string'
+        },
+    },
+    'required': ['url'],
+    'additionalProperties': False
+}
+
+FFMPEG = '/opt/bin/ffmpeg'
+
+# Lambda has a 6MB limit for request and response, however, base64 adds 33% overhead.
+# Also, leave a few KB for the headers.
+MAX_VIDEO_SIZE = 6 * 1024 * 1024 * 3 // 4 - 4096
```

Comment:
What is the logic behind 4.1 kB for headers?

Reply:
Kind of arbitrary - but it's pretty unlikely for the headers to take up more than 4KB. When I try it in a browser, it's just under 1KB.